### ISSUE TITLE
(LowerPriority) enhance detailing by allowing image "map"

### ISSUE THREAD

**Issue by jlsjonas** [TWiStErRob/TWiStErRob#44]
f.e. a closet with multiple parts, clicking the part would be easier then finding it in a (possibly bigger) list, or would require more splitting

**Comment by TWiStErRob** [comment]
The current approach to tackle huge lists is to create sub-items with the compartments category and also Group is special category that has a different UI if it has stuff in it. Just like file folders (a concept widespread and known already)

That said, associating parts of the main image with sub-items freely may be useful, but it's a lot of work by the user to create the map.

**Comment by jlsjonas** [comment]
Hence also indicating low priority (would've used label if I could)
it's something that could be worth the effort, depending on the environment and use case.
But it's something worth considering for the future.

**Comment by TWiStErRob** [comment]
Showing the data to the user is just a tap-map for the detail image, but the editor for the map is a good chunk of new UI work as well, that's why I think it's a feature. Also it would be listed under the features on the store description...

**Comment by jlsjonas** [comment]
Yes, sorry I worded that poorly, it's definitely a new feature, I just meant it shouldn't replace the "current" way, except if f.e. the tap-map is complete (all children are marked & sized big enough to be easily pressed)

**Comment by TWiStErRob** [comment]
> all children are marked & sized big enough to be easily pressed

Yeah, sure, like that will ever happen... remember, all user input is evil ;)

**Comment by jlsjonas** [comment]
hence also mentioning it ;) the area of each tap-grid should be checked and be a certain minimum, items that are too small should be listed as before, prioritized over the "big enough" items

**Comment by TWiStErRob** [comment]
Considering the minimum recommended tap area being some 48-56dp there aren't many subitem that can be linked this way with the current header size.

**Comment by jlsjonas** [comment]
correct, but that image should be taking pretty much the entire screen, and/or having it partially zoomed in (matching screen size, allowing "scrolling" the image only bidirectional, not omni)

**Comment by TWiStErRob** [comment]
Zooming is a good idea.

Though note that currently it sounds like this would replace/augment the current UX of scrolling an alphabetized list to scrolling/zooming an unstructured image. But I agree, the less reading to find an item the better, and images do just that.

**Comment by jlsjonas** [comment]
> it sounds like this would replace/augment the current UX

in theory you could have: 
- [ ] left/right scrolling over the image (if landscape mode picture viewed in portrait)
- [ ] swiping up to reveal the "current" way
- [ ] swiping down to show the default (since you won't be able to swipe left for that anymore)

**Comment by TWiStErRob** [comment]
I feel that what you described just now would make the UX inconsistent for item. Some items would display the current way and some (very few) would show unexpected UI. Also swiping down is pretty much reserved for pull to refresh.

That said a bottom sheet showing expanded by default might help here. The bottom sheet would contain this extreme navigation method of zooming/panning and in this case, a swipe down would actually be the right gesture to get to the "raw data" behind the picture (i.e. collapse the bottom sheet and reveal the traditional UI).

**Comment by jlsjonas** [comment]
> would make the UX inconsistent for items

In a way yes, though when this feature gets implemented that UX could be made consistent.

> Also swiping down is pretty much reserved for pull to refresh.

In cases where it makes sense: definitely, in my example I would still assume a "hint" towards the extra content, which shouldn't make the swipe down gesture break the design guides

**Comment by TWiStErRob** [comment]
Blueprint: https://mail.google.com/mail/u/0/#inbox/15f17b88b5ba4566
